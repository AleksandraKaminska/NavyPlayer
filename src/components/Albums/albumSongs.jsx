import React, { Component } from 'react'
import { connect } from 'react-redux'
import fetchJsonp from 'fetch-jsonp'
import store from 'store'
import * as actions from 'actions';
import { randomAlbumTrack, searchArtistInfo } from 'helperFunctions'
import { NavLink } from 'react-router-dom';

const { DZ } = window

class AlbumSongs extends Component {
    findAlbum = () => {
        store.dispatch(actions.changePlaylistAction(0))
        store.dispatch(actions.changeArtistPlaylistAction([]))
        randomAlbumTrack(this.props)
    }

    fetchData = id => {
        return dispatch => fetchJsonp(`https://api.deezer.com/track/${id}?output=jsonp`)
        .then(response => response.json())
        .then(data => dispatch(actions.changeTrackAction(data)))
    }

    handleClick = id => {
        const { track } = this.props
        store.dispatch(actions.prevTrackAction(track))
        store.dispatch(this.fetchData(id))
        .then(({ track }) => {
            DZ.player.pause()
            searchArtistInfo(track)
            DZ.player.playTracks([track.id])
        })
    }

    render() {
        const { tracks, id } = this.props.album
        let songs = tracks && tracks.data.map(({ id, title }, i) => (
            <li onClick={() => this.handleClick(id)} key={i}>
                {i < 9 ? '0' : null}{i + 1}. {title}
            </li>
        ))

        return (
            <div
                className="songs"
                style={window.innerWidth >= 870 ? { width: '25em', right: '-27em' } : { width: '100%' }}
                ref={this.props.songs}
            >
                <NavLink
                    to={`/album/${id}`}
                    onClick={this.findAlbum}
                    className="playAlbum"
                    style={{ visibility: songs && 'visible' }}
                >
                    Play album
                </NavLink>
                {songs && <ul>{songs}</ul>}
            </div>
        )
    }
}


const mapStateToProps = ({ track, album }) => ({ track, album })

export default connect(mapStateToProps)(AlbumSongs)
