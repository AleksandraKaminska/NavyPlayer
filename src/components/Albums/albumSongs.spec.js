import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { AlbumSongs } from './albumSongs'
import renderer from 'react-test-renderer'

describe('AlbumSongs', () => {
  it('should render self', () => {
    const rendered = renderer.create(
      <Router>
        <AlbumSongs album={{}} track={{}} />
      </Router>
    )
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
