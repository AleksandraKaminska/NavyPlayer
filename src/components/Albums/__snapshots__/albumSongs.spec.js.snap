// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumSongs should render self 1`] = `
<div
  className="songs"
  style={
    Object {
      "right": "-27em",
      "width": "25em",
    }
  }
>
  <a
    aria-current={null}
    className="playAlbum"
    href="/album/undefined"
    onClick={[Function]}
    style={
      Object {
        "visibility": undefined,
      }
    }
  >
    Play album
  </a>
</div>
`;
